crypto/pem/pem_lib.c

```diff
@@ -362,7 +362,8 @@ int PEM_ASN1_write_bio(i2d_of_void *i2d, const char *name, BIO *bp,
             kstr = (unsigned char *)buf;
         }
         RAND_add(data, i, 0);   /* put in the RSA key. */
-        OPENSSL_assert(EVP_CIPHER_iv_length(enc) <= (int)sizeof(iv));
+        if (!ossl_assert(EVP_CIPHER_iv_length(enc) <= (int)sizeof(iv)))
+            goto err;
```

Comment:
After additional consideration in light of previous reasoning around hmac, ot probably has to be treated as run-time condition. [Because just like with hmac, it's not given that (currently non-existent) check in evp/cmeth_lib.c would be equivalent of this one.] But I reckon that position of this check is questionable. It appears that it would be more appropriate to move it to the very beginning of the function and merge it with PEM_R_UNSUPPORTED_CIPHER.